cmd/control/control.go

```diff
@@ -59,6 +68,38 @@ func initialiseDatabase(conf config.Database) (database.Database, error) {
 	}
 }
 
+func AuthenticatorFromConfig(config config.ControlConfiguration) configFileAuthenticator {
```

Comment:
This is racy! If one request creates a new auth token, while another read it, you'll read from a concurrently updating map, which is bad!

Either this needs a mutex, or we should add a mutex to `femto.Femto` for the calls to `Authenticator`. I think I'd prefer the former, as we don't want to lock the mutex when doing hashing (when we do that).

Before doing this, right a test for it. See 0401d4c57b492bde063c38b71477df9e7f2a5de6 for an example of that.

Reply:
Thanks for pointing this out, kinda forgot that I was writing concurrent code!